openhtf/output/web_gui/src/update_plugins.py

```diff
@@ -0,0 +1,248 @@
+<!-- THIS IS AUTO-GENERATED, IF YOU WANT TO EDIT GO TO update_plugins.py to edit -->
+<!-- ........................................................................... -->
+<div id="station">
+  <station-header
+    [stationInfo]="stationInfo"
+    [plugins]="plugins">
+  </station-header>
+  <div class="currentTab" id="current">
+    <div *ngIf="tests.length == 0" class="big-message">
+      No tests available for display.
+    </div>
+    <div *ngFor="let test of tests">
+      <div class="test">
+        <test-header
+          [dutID]="test.test_record?.dut_id"
+          [startTime]="test.test_record?.start_time_millis"
+          [endTime]="test.test_record?.end_time_millis"
+          [currentMillis]="currentMillis"
+          [status]="test.test_record?.outcome
+                    || test.status || 'OFFLINE'">
+        </test-header>
+        <div class="test-content">
+          <phase-listing id="phases" [state]="test"
+                                     [currentMillis]="currentMillis">
+          </phase-listing>
+          <logs id="logs" [log_records]="test.test_record?.log_records">
+          </logs>
+          <metadata id="metadata" [metadata]="test.test_record.metadata">
+          </metadata>
+        </div>
+      </div>
+    </div>
+  </div>
+'''
+post_html='''</div>'''
+
+pre_ts="""// Copyright 2016 Google Inc.All Rights Reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/////////////////////////////////////////////////////////////////////////
+//AUTOMATICALLY GENERATED - IF YOU WANT TO EDIT GO TO update_plugins.py//
+/////////////////////////////////////////////////////////////////////////
+
+declare var $;  // Global provided by the jquery package.
+
+import {Component,
+        OnDestroy,
+        OnInit,
+        SimpleChange} from 'angular2/core';
+import {RouteParams} from 'angular2/router';
+import {RouterLink} from 'angular2/router';
+
+import {Logs} from './station.logs';
+import {StationHeader} from './station.header';
+import {Metadata} from './station.metadata';
+import {PhaseListing} from './station.phases';
+import {Prompt} from './station.prompt';
+import {StationService} from './station.service';
+import {TestHeader} from './station.testheader';
+import {Countdown, ObjectToArray} from './utils';
+import 'file?name=/styles/station.css!./station.css';
+import 'file?name=/templates/station.html!./station.html';
+"""
+
+mid1_ts='''
```

Comment:
Instead of cutting this up in such an odd location (directives opens but doesn't close the list), put a format directive. Also, use `"""` when possible

```py
mid1_ts = """
@Component({
  selector: 'station',
  ...
  directives: [StationHeader, Prompt, ..., Metadata, {extra_directives}],
  providers: [StationService]
})
export class ...
"""
```